src/gui.cc

```diff
@@ -541,13 +541,13 @@ void GUI::DisplayTimeline(
             bool item_present = false;
             TogglTimelineEventView *event_app = first_event;
             while (event_app) {
-                if (compare_string(event_app->Filename, event.Filename().c_str()) == 0) {
+                if (compare_string(event_app->Filename, copy_string(event.Filename().c_str())) == 0) {
```

Comment:
`copy_string` here is completely unnecessary (just `c_str` is fine) and you don't free the result anywhere, causing a considerable memory leak. Same for the usages in the same file below.

Reply:
Unfortunately, c_str() causes the error in Windows. As I remember that it was
```
Error | C2664 | 'int compare_string(const wchar_t *,const wchar_t *)': cannot convert argument 2 from 'const _Elem *' to 'const wchar_t *'        with        [            _Elem=char        ] | TogglDesktopDLL | C:\Users\nghiatran\Desktop\toggldesktop\src\gui.cc | 544 |  
```

<img width="1348" alt="Screen Shot 2019-11-20 at 20 38 56" src="https://user-images.githubusercontent.com/5878421/69243508-cd635480-0bd5-11ea-80e9-0d97b1f50e58.png">

It's reason why I try using `copy_string`.

Reply:
👍 for the point regarding the leaking on c_string. I will fix it now

Reply:
Using `copy_string` is still wrong in this case unless you use `free` to the returned pointer afterwards (which introduces an unnecessary performance overhead).
The way to right way to fix this on the Windows side is to use the same code we have in `copy_string` to convert `std::string` to `std::wstring`:
```
char_t *copy_string(const std::string &s) {
#if defined(_WIN32) || defined(WIN32)
    std::wstring ws;
    Poco::UnicodeConverter::toUTF16(s, ws);
    return wcsdup(ws.c_str());
#else
    return strdup(s.c_str());
#endif
}
```
The part between the `#if` and `#else` is what you need to use to convert the string to a wide string.

Reply:
Thank for the hint. I will try this approach 🌮

Reply:
It actually seems `to_char_t` in `toggl_api_private.cc` does the same thing but I'm not entirely sure about that. 
It also seems to be written in a very dangerous way (but it probably still works most of the time)... This is a different issue and I'll open a report for it.

Reply:
I choose the safe way to release the `char_t *` after using the `copy_string` rather than fixing the the `copy_string` body since it could affect entire the library. Please help me to review on the second round @MartinBriza 👍